spock-core/src/main/java/org/spockframework/runtime/model/FeatureInfo.java

```diff
@@ -119,6 +126,27 @@ public void addDataProvider(DataProviderInfo dataProvider) {
     dataProviders.add(dataProvider);
   }
 
+  public List<FeatureInfo> getDependees() {
+    return dependees;
+  }
+
+  public List<FeatureInfo> getDependencies() {
+    return dependencies;
+  }
+
+  public void addDependency(FeatureInfo featureInfo) {
```

Comment:
That's indeed correct.

I would indeed not impose any implicit ordering due to the dependency configuration.

This could easily conflict with explicitly defined order from another extension and actually the way it is right now it can also be used for other use-cases like "if this feature is executed, this other feature also always needs to be executed".

This could for example also be used for something like having a feature that does a cleanup for another feature. Then the cleanup feature would be the dependency but configured to run after the dependee.
Or for similar use-cases, whatever a user might come up with. :-D

The meaning is just that if the dependee is run, the dependency will also run, no matter whether before, after, or concurrently, and I think it would be best to keep it like that.

So what do you suggest?
Adding JavaDoc that explains that it does not impose order, just that it is run too?
Or a different name?
If the latter, any idea for good replacements for `dependee` and `dependency`?

Reply:
> This could for example also be used for something like having a feature that does a cleanup for another feature.

That sounds like a terrible idea 😅.

I think it needs a JavaDoc in any case, explaining what it does and doesn't do. As for a name, maybe `addRequiredExecution` or `addImpliedExecution`? Not sure if those are better.

Reply:
We should also make sure that you can only depend on features in the same Spec hierarchy, it don't think cross spec relationships is something we should support.

Reply:
Why not, as long as it does not impose order?

Reply:
I think coupling between specs is a bad idea.

Reply:
Tbh, meanings about what is a bad idea and what isn't greatly vary from person to person.
I'm not too much of a fan of forbidding something that would work perfectly fine without any reason besides personal opinion.

Reply:
I renamed them to implied and implying features and added JavaDoc and `@Beta`